gpx.cc

```diff
@@ -264,24 +246,23 @@ gpx_reset_short_handle()
 }
 
 static void
-gpx_write_gdata(gpx_global_entry* ge, const char* tag)
+gpx_write_gdata(const QStringList& ge, const char* tag)
 {
-  queue* elem, *tmp;
-
-  if (!gpx_global || QUEUE_EMPTY(&ge->queue)) {
-    return;
-  }
-  writer->writeStartElement(tag);
-  QUEUE_FOR_EACH(&ge->queue, elem, tmp) {
-    gpx_global_entry* gep = BASE_STRUCT(elem, gpx_global_entry, queue);
-    writer->writeCharacters(gep->tagdata);
-    /* Some tags we just output once. */
-    if ((0 == strcmp(tag, "url")) ||
-        (0 == strcmp(tag, "email"))) {
-      break;
+  if (!ge.isEmpty()) {
+    writer->writeStartElement(tag);
+    // TODO: this seems wrong, it is concatenating element content
+    // from multiple illegally repeated elements into one.
+    // However, this does result in output that complies with the schema.
+    foreach (const QString& str, ge) {
```

Comment:
for (auto &str: ge) { ?

Reply:
foreach (auto& str, ge)
works and preserves const, but I haven't pursued auto too much.
clang-tidy modernize-use-auto misses this, probably because foreach is a qtism.